src/tokens/Assets.tsx

```diff
@@ -0,0 +1,442 @@
+  const displayPositions = showPositions && positions.length > 0
+
+  const dappShortcutsEnabled = getFeatureGate(StatsigFeatureGates.SHOW_CLAIM_SHORTCUTS)
+  const positionsWithClaimableRewards = useSelector(positionsWithClaimableRewardsSelector)
+  const showClaimRewards =
+    dappShortcutsEnabled &&
+    positionsWithClaimableRewards.length > 0 &&
+    activeTab !== AssetTabType.Collectibles
+
+  // TODO: Update these to filter out unsupported networks once positions support non-Celo chains
+  const totalPositionsBalanceUsd = useSelector(totalPositionsBalanceUsdSelector)
+  const totalPositionsBalanceLocal = useDollarsToLocalAmount(totalPositionsBalanceUsd)
+  const totalBalanceLocal = totalTokenBalanceLocal?.plus(totalPositionsBalanceLocal ?? 0)
+
+  const [nonStickyHeaderHeight, setNonStickyHeaderHeight] = useState(0)
+  const [listHeaderHeight, setListHeaderHeight] = useState(0)
+  const [listFooterHeight, setListFooterHeight] = useState(0)
+
+  const scrollPosition = useSharedValue(0)
+  const footerPosition = useSharedValue(0)
+  const handleScroll = useAnimatedScrollHandler<{ prevScrollY: number }>(
+    {
+      onScroll: (event, ctx) => {
+        const scrollY = event.contentOffset.y
+        scrollPosition.value = scrollY
+
+        function clamp(value: number, min: number, max: number) {
+          return Math.min(Math.max(value, min), max)
+        }
+
+        // Omit overscroll in the calculation
+        const clampedScrollY = clamp(
+          scrollY,
+          0,
+          event.contentSize.height - event.layoutMeasurement.height
+        )
+
+        // This does the same as React Native's Animated.diffClamp
+        const diff = clampedScrollY - ctx.prevScrollY
+        footerPosition.value = clamp(footerPosition.value + diff, 0, listFooterHeight)
+        ctx.prevScrollY = clampedScrollY
+      },
+      onBeginDrag: (event, ctx) => {
+        ctx.prevScrollY = event.contentOffset.y
+      },
+    },
+    [listFooterHeight]
+  )
+
+  const animatedListHeaderStyles = useAnimatedStyle(() => {
+    if (nonStickyHeaderHeight === 0) {
+      return {
+        shadowColor: 'transparent',
+        transform: [
+          {
+            translateY: -scrollPosition.value,
+          },
+        ],
+      }
+    }
+
+    return {
+      transform: [
+        {
+          translateY:
+            scrollPosition.value > nonStickyHeaderHeight
+              ? -nonStickyHeaderHeight
+              : -scrollPosition.value,
+        },
+      ],
+      shadowColor: interpolateColor(
+        scrollPosition.value,
+        [nonStickyHeaderHeight - 10, nonStickyHeaderHeight + 10],
+        ['transparent', 'rgba(48, 46, 37, 0.15)']
+      ),
+    }
+  }, [scrollPosition.value, nonStickyHeaderHeight])
+
+  const animatedFooterStyles = useAnimatedStyle(() => {
+    return {
+      transform: [
+        {
+          translateY: footerPosition.value,
+        },
+      ],
+    }
+  }, [footerPosition.value])
+
+  useScrollAwareHeader({
```

Comment:
I've never used `useScrollAwareHeader` before -- what purpose does this serve? Basically to just have a header that hides when the user scrolls?